src/components/ProgressArray.tsx

```diff
@@ -8,18 +8,16 @@ import {
 import ProgressCtx from "./../context/Progress";
 import GlobalContext from "./../context/Global";
 import StoriesContext from "./../context/Stories";
+import { timestamp } from "../util/time";
 
 export default () => {
   const [count, setCount] = useState<number>(0);
-  const { currentId, next, videoDuration, pause, bufferAction } = useContext<ProgressContext>(
-    ProgressCtx
-  );
-  const {
-    defaultInterval,
-    onStoryEnd,
-    onStoryStart,
-    onAllStoriesEnd,
-  } = useContext<GlobalCtx>(GlobalContext);
+  const lastTime = useRef<number>(timestamp());
+
+  const { currentId, next, videoDuration, pause, bufferAction } =
+    useContext<ProgressContext>(ProgressCtx);
+  const { defaultInterval, onStoryEnd, onStoryStart, onAllStoriesEnd } =
+    useContext<GlobalCtx>(GlobalContext);
   const { stories } = useContext<StoriesContextInterface>(StoriesContext);
 
   useEffect(() => {
```

Comment:
You can add a initializer for the timestamp in this useEffect so that you are sure it run only on the client.